ruma-federation-api/src/membership/create_invite/v1.rs

```diff
@@ -0,0 +1,121 @@
+//! [PUT /_matrix/federation/v1/invite/{roomId}/{eventId}](https://matrix.org/docs/spec/server_server/r0.1.4#put-matrix-federation-v1-invite-roomid-eventid)
+
+use js_int::UInt;
+use ruma_api::ruma_api;
+use ruma_events::{room::member::MemberEventContent, EventType};
+use ruma_identifiers::{EventId, RoomId, ServerName, UserId};
+use serde::{Deserialize, Serialize};
+
+use super::{InviteEvent, StrippedState};
+
+ruma_api! {
+    metadata: {
+        description: "Invites a remote user to a room.",
+        method: PUT,
+        name: "create_invite",
+        path: "/_matrix/federation/v1/invite/:room_id/:event_id",
+        rate_limited: false,
+        requires_authentication: true,
+    }
+
+    #[non_exhaustive]
+    request: {
+        /// The room ID that the user is being invited to.
+        #[ruma_api(path)]
+        pub room_id: RoomId,
+
+        /// The event ID for the invite event, generated by the inviting server.
+        #[ruma_api(path)]
+        pub event_id: EventId,
+
+        /// The matrix ID of the user who sent the original `m.room.third_party_invite`.
+        pub sender: UserId,
+
+        /// The name of the inviting homeserver.
+        pub origin: Box<ServerName>,
+
+        /// A timestamp added by the inviting homeserver.
+        pub origin_server_ts: UInt,
+
+        /// The value `m.room.member`.
+        #[serde(rename = "type")]
+        pub kind: EventType,
+
+        /// The user ID of the invited member.
+        pub state_key: UserId,
+
+        /// The content of the event.
+        pub content: MemberEventContent,
+
+        /// Information included alongside the event that is not signed.
+        pub unsigned: UnsignedEventContent,
+    }
+
+    #[non_exhaustive]
+    response: {
+        /// The response invite event
+        #[ruma_api(body)]
+        #[serde(with = "crate::serde::invite_response")]
+        invite_response: InviteEvent,
```

Comment:
Maybe just name this `event` too? Since it's `#[ruma_api(body)]` its name doesn't matter, but `response.invite_response` doesn't seem so nice.

Reply:
That does seem a lot nicer, I forgot the attribute means the name doesn't matter